src/execution_plan/ops/op_merge_create.c

```diff
@@ -102,7 +102,16 @@ static bool _CreateEntities(OpMergeCreate *op, Record r) {
 		/* Convert query-level properties. */
 		PropertyMap *map = n->properties;
 		PendingProperties *converted_properties = NULL;
-		if(map) converted_properties = ConvertPropertyMap(r, map);
+		if(map) {
+			converted_properties = ConvertPropertyMap(r, map);
+			for(int i = 0; i < converted_properties->property_count; i++) {
+				if(SI_TYPE(converted_properties->values[i]) & T_NULL) {
+					PendingPropertiesFree(converted_properties);
+					QueryCtx_SetError("Cannot merge node using null property value");
+					QueryCtx_RaiseRuntimeException();
+				}
+			}
```

Comment:
Isn't this redundant ?
`ConvertPropertyMap` in `src/execution_plan/ops/shared/create_functions.c`  performs this validation already.

Reply:
I don't know what the justification for these differing behaviors are, but:
```
CREATE (a:L {v1: NULL, v2: 'prop'}) RETURN a
```
Creates one node with the property `v2`, while:
```
MERGE (a:L {v1: NULL, v2: 'prop'}) RETURN a
```
Is supposed to generate this error.